lib/src/embedded/dispatcher.ts

```diff
@@ -77,24 +74,43 @@ export class Dispatcher {
 
   constructor(
     private readonly outboundMessages$: Observable<OutboundTypedMessage>,
-    private readonly writeInboundMessage: (message: InboundTypedMessage) => void
+    private readonly writeInboundMessage: (
+      message: InboundTypedMessage
+    ) => void,
+    private readonly handleImportRequest: (
+      request: OutboundMessage.ImportRequest
+    ) => PromiseOr<InboundMessage.ImportResponse>,
+    private readonly handleFileImportRequest: (
+      request: OutboundMessage.FileImportRequest
+    ) => PromiseOr<InboundMessage.FileImportResponse>,
+    private readonly handleCanonicalizeRequest: (
+      request: OutboundMessage.CanonicalizeRequest
+    ) => PromiseOr<InboundMessage.CanonicalizeResponse>,
+    private readonly handleFunctionCallRequest: (
+      request: OutboundMessage.FunctionCallRequest
+    ) => PromiseOr<InboundMessage.FunctionCallResponse>
```

Comment:
Consider making these named parameters. Otherwise, a bunch of functions of similar types all in a row can get pretty confusing.